### PR TITLE
Feat/get cl config from node

### PR DESCRIPTION
**Pull request by vgorkavenko**

### CODE REVIEW

**Review by F4ever — COMMENTED** [review]
Remove fixtures + merge conflicts

**Comment by vgorkavenko** [comment]
> Remove fixtures + merge conflicts

done

File: src/providers/consensus/client.py [hunk 1]
```diff
@@ -39,7 +39,13 @@ class ConsensusClient(HTTPProvider):
     API_GET_SPEC = 'eth/v1/config/spec'
     API_GET_GENESIS = 'eth/v1/beacon/genesis'
 
-    def get_config_spec(self):
+    @property
+    def spec(self):
```
Comment (F4ever) [inline src/providers/consensus/client.py:43]: annotation

File: src/providers/consensus/client.py [hunk 2]
```diff
@@ -39,7 +39,13 @@ class ConsensusClient(HTTPProvider):
...
+        """Consensus Layer spec"""
+        return self._get_config_spec()
+
+    @lru_cache(maxsize=1)
+    def _get_config_spec(self):
```
Comment (F4ever) [inline src/providers/consensus/client.py:48]: annotation

File: src/providers/consensus/typings.py [hunk 3]
```diff
@@ -11,6 +11,24 @@ class BeaconSpecResponse(FromResponse):
     DEPOSIT_CHAIN_ID: str
     SLOTS_PER_EPOCH: str
     SECONDS_PER_SLOT: str
+    # https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#time-parameters-1
+    MIN_VALIDATOR_WITHDRAWABILITY_DELAY: str
```
Comment (F4ever) [inline src/providers/consensus/typings.py:15]: Maybe lets parse response to expected value

Comment (vgorkavenko) [reply src/providers/consensus/typings.py:15]: We don't do it in other requests. Why should we do it here?

File: src/modules/ejector/ejector.py [hunk 4]
```diff
@@ -241,26 +235,32 @@ def _get_predicted_withdrawable_epoch(
...
-        epochs_required_to_exit_validators = (validators_to_eject_count - remain_exits_capacity_for_epoch) // churn_limit + 1
+        epochs_required_to_exit_validators = (
+            (validators_to_eject_count - remain_exits_capacity_for_epoch) // churn_limit + 1
+        )
 
-        return EpochNumber(max_exit_epoch_number + epochs_required_to_exit_validators + MIN_VALIDATOR_WITHDRAWABILITY_DELAY)
+        return EpochNumber(
+            max_exit_epoch_number +
+            epochs_required_to_exit_validators +
+            int(self.w3.cc.spec.MIN_VALIDATOR_WITHDRAWABILITY_DELAY)
```
Comment (F4ever) [inline src/modules/ejector/ejector.py:253]:
```suggestion
            self.w3.cc.spec.MIN_VALIDATOR_WITHDRAWABILITY_DELAY
```

File: src/modules/ejector/ejector.py [hunk 5]
```diff
@@ -241,26 +235,32 @@ def _get_predicted_withdrawable_epoch(
...
     @staticmethod
-    def compute_activation_exit_epoch(blockstamp: ReferenceBlockStamp):
+    def compute_activation_exit_epoch(spec: BeaconSpecResponse, blockstamp: ReferenceBlockStamp) -> EpochNumber:
         """
         Return the epoch during which validator activations and exits initiated in ``epoch`` take effect.
 
         Spec: https://github.com/LeastAuthority/eth2.0-specs/blob/dev/specs/phase0/beacon-chain.md#compute_activation_exit_epoch
         """
-        return blockstamp.ref_epoch + 1 + MAX_SEED_LOOKAHEAD
+        return EpochNumber(blockstamp.ref_epoch + 1 + int(spec.MAX_SEED_LOOKAHEAD))
```
Comment (F4ever) [inline src/modules/ejector/ejector.py:263]:
```suggestion
        return EpochNumber(blockstamp.ref_epoch + 1 + spec.MAX_SEED_LOOKAHEAD)
```

**Pull request merged**